DEVELOPING.rst

```diff
@@ -0,0 +1,237 @@
+Developing
+==========
+
+.. _developing-reference:
+
+Architecture Overview
+---------------------
+
+The high-level architecture of `renku` follows a Hexogonal/Ports&Adapters
```

Comment:
I'm not sure what single backquotes do in RST (it does nothing in GH; it makes the text italic in PyCharm). You need to replace \` with \`\` if you want to show them as code blocks: \`renku\` -> \`\`renku\`\`. This happens through the whole document (except when it is preceded by \:code\: which shows up as code block).

Reply:
It makes it italic or shows it as a code block. Actually it's not defined and implementation specific. 
From the spec:
> The rendering and meaning of interpreted text is domain- or application-dependent. It can be used for things like index entries or explicit descriptive markup (like program identifiers). 

For instance in our docstrings in code: 
\`\`renku\`\` -> autodoc would try and find a reference to `renku` and throw a warning if such a reference doesn't exist.
\`renku\` -> just treated as a literal, not used for cross referencing, not checked by the spell checker
renku -> normal text, might fail a spellcheck

So doing the whole docstrings thing a while back taught me that single backtick is safe. But of course that's only for code docstrings, and you're completely right that it doesn't make sense here.